vterm/csi.go

```diff
@@ -10,23 +10,21 @@ func (v *VTerm) handleEraseInDisplay(directive int) {
 	switch directive {
 	case 0: // clear from Cursor to end of screen
 		for i := v.Cursor.X; i < len(v.Screen[v.Cursor.Y]); i++ {
```

Comment:
Is `len(v.Screen[v.Cursor.Y])` just width (looking at [vterm.go](https://github.com/aaronjanse/3mux/blob/e30fbe753dd560c4c02b46ffbf735bdde04b427a/vterm/vterm.go), it seems that this is the case? I could be completely wrong though)?

Reply:
When a vterm is resized to a smaller width, it doesn't delete columns from `v.Screen`, so `v.Screen` may have more columns than `v.w`. This allows shrinking-then-expanding panes without characters disappearing.

Future versions of 3mux should probably use wrapping to show columns past `v.w`, although I'm not sure how that would be implemented.

https://github.com/aaronjanse/3mux/blob/e30fbe753dd560c4c02b46ffbf735bdde04b427a/vterm/vterm.go#L106-L140

Reply:
Also, vterm currently has a race condition between drawing code and resizing. I can reproduce it via fuzzing but not while interactively using 3mux. I think race condition could be fixed by pausing rendering while resizing.

Reply:
For wrapping, if we used row/col-major for the screen, we can simply adjust a screenWidth / screenHeight and iterate through the screen which would basically be a 1D array:
[ a, b, c, d, e, f]

2 x 3 screen:
```
abc
def
```
3 x 2 screen:
```
ab
cd
ef
```

Though, the loops that use screen would have to change to adjust for row/col-major.

Reply:
I assume you mean row-major, based on the example? Regardless, I do think we need to eventually remember wrapping properly, and using really long rows might be the way to go, with some fancy index tracking stuff